e2e_playwright/shared/performance.py

```diff
@@ -121,12 +123,18 @@ def measure_performance(
         captured_traces = captured_traces_result.get("value", "{}")
         parsed_captured_traces = json.loads(captured_traces)
 
+        performance_results_dir = os.path.join(
```

Comment:
nit: another trick we are using in some similar situations (e.g. to access test assets) is to resolve the path relative to a given python script file:

https://github.com/streamlit/streamlit/blob/b4feb8e9b38b34f74618b9d471a4785392db5052/e2e_playwright/st_chat_message.py#L23

But no strong opinion either way as long as it works locally and in CI.

Reply:
Yeah, I've gone back and forth on this. I like the idea of actually reading the info from git since no changes will need to be made if the file gets moved around on the file system. I understand that's probably not something that will happen often, but it's nice to not have to worry about it when eventual refactors happen.

Also just to confirm this does work locally and in CI. It was able to successfully find all `__snapshots__` and the artifact outputs are correct: https://github.com/streamlit/streamlit/actions/runs/12398270107?pr=10039 I also proactively added a `raise` statement in case there are issues with issuing commands to `git`.